### PR TITLE
improve error message when trying to 'git secret add' a file tracked by git

### LINKED ISSUE #126: git secret adding a tracked file fails with ambiguous message "file_name is not a file. abort"
What are the steps to reproduce this issue?
-------------------------------------------
1. `git secret add -i already_tracked_filename`

What happens?
-------------
File not added, operation aborted.

What were you expecting to happen?
----------------------------------
- Option 1: File to be added to the "to encrypt" files list.
Or 
- Option 2: See a meaningful message telling that tracked files can't be added to the list and advise me to untrack the file myself with a `git rm --cached already_tracked_filename`

Any logs, error output, etc?
----------------------------
`already_tracked_filename is not a file. abort.`

Any other comments?
-------------------
@sobolevn which option do you prefer 1 or 2? I'm asking because I'm interested in working on it (if I find time).

What versions of software are you using?
----------------------------------------
**Operating system:** (`uname -a`) …
`Linux instance-2-git-secret 4.9.0-4-amd64 #1 SMP Debian 4.9.65-3 (2017-12-03) x86_64 GNU/Linux`

**`git-secret` path:** (`which git-secret`) …
`/usr/local/bin/git-secret`

**`git-secret` version:** (`git secret --version`) …
`0.2.2`

**`git` version:** (`git --version`) …
`git version 2.11.0`

**Shell type and version:** (`$SHELL --version`) …
`GNU bash, version 4.4.12(1)-release (x86_64-pc-linux-gnu)`

**`gpg` version:** (`gpg --version`) …
```
gpg (GnuPG) 2.1.18
libgcrypt 1.7.6-beta
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /home/sayhi_aymen/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```

### PR DESCRIPTION
**Pull request by joshrabinowitz**
Closes #126 

Now complains "file tracked in git, consider using 'git rm --cached yourfilename.txt'"

### CODE REVIEW

File: src/commands/git_secret_add.sh [hunk 1]
```diff
@@ -32,9 +32,14 @@ function add {
     normalized_path=$(_git_normalize_filename "$item")
     path=$(_append_root_path "$normalized_path")
 
+    # check that the file is not tracked
+    if [[ $(git ls-files --error-unmatch "$item") ]]; then
```
Comment (sobolevn) [inline src/commands/git_secret_add.sh]: can we move this to a separate function?

**Comment by sobolevn** [comment]
Nice!

P.S. You can use https://help.github.com/articles/closing-issues-using-keywords/ That's quite useful.

**Comment by joshrabinowitz** [comment]
If you try to add a file that is tracked by git, you'll get output like this:

    % git secret add normalfile.txt
    git-secret: abort: file tracked in git, consider using 'git rm --cached normalfile.txt'

File: src/commands/git_secret_add.sh (hunk 1 above)
Comment (joshrabinowitz) [reply src/commands/git_secret_add.sh]: done!

**Comment by joshrabinowitz** [comment]
I think this is ready to merge

File: src/commands/git_secret_add.sh (hunk 1 above)
Comment (aymens) [reply src/commands/git_secret_add.sh]: @joshrabinowitz thank you for doing this.
@sobolevn my apologies, didn't manage to find time.

I've a comment/question, the option **--error-unmatch** would display an error message for untracked files, like following:
```
error: pathspec 'untracked_file' did not match any file(s) known to git.
Did you forget to 'git add'?
```
Don't you think that the option **--cached** would be more appropriate ?
```
if [[ $(git ls-files --cached "$item") ]]; then
...
```

Comment (joshrabinowitz) [reply src/commands/git_secret_add.sh]: @aymens  

That's not the error that happens in that case. 

If the file is completely new they new get the error:

````
% git secret add new_file_not_in_git.txt
git-secret: abort: these files are not in .gitignore: new_file_not_in_git.txt
````

If the new file is `git add`ed they get:
````
    % git secret add file_added_to_git.txt
    git-secret: abort: file tracked in git, consider using 'git rm --cached file_added_to_git.txt'
````

Comment (joshrabinowitz) [reply src/commands/git_secret_add.sh]: and thanks for chiming in @aymens

Comment (aymens) [reply src/commands/git_secret_add.sh]: That's fine then!

**Pull request merged**